build.go

```diff
@@ -89,65 +74,52 @@ func main() {
 	case "6", "7":
 		dsms = append(dsms, int((*dsmVersion)[0]-'0'))
 	case "all":
-		if doBuild {
-			log.Fatalf("invalid --dsm=all in --version=build mode")
-		}
 		dsms = append(dsms, 6, 7)
 	}
-	if *goarch == "all" && doBuild {
-		log.Fatalf("invalid --goarch=all in --version=build mode")
-	}
-	if strings.HasSuffix(*output, ".spk") && !doBuild {
-		log.Fatalf("-o value of *.spk only supported in --version=build mode")
-	}
-
-	shortVer := *version
-	if doBuild {
-		var err error
-		shortVer, err = getShortVer(*srcDir)
-		if err != nil {
-			log.Fatal(err)
+	var archs []string
+	if *goarch == "all" {
+		for goarch := range synPlat {
+			archs = append(archs, goarch)
 		}
-	}
-	var synoArch string
-	if vv := synPlat[*goarch]; len(vv) > 0 {
-		synoArch = vv[0]
 	} else {
-		log.Fatalf("unknown --goarch value %q", *goarch)
-	}
-	param := spkParams{
-		createTime:       time.Now(),
-		version:          shortVer,
-		spkBuildBase:     *spkBuild,
-		goarch:           *goarch,
-		synoArch:         synoArch,
-		dsm:              dsms[0],
-		forPackageCenter: false,
-		srcDir:           *srcDir,
-	}
-
-	file := param.filename()
-	out := param.outputFile()
-	log.Printf("Generating %v ...", file)
-	var buf bytes.Buffer
-	if err := genSPK(&buf, param); err != nil {
-		log.Fatal(err)
+		if vv := synPlat[*goarch]; len(vv) == 0 {
+			log.Fatalf("unknown --goarch value %q", *goarch)
+		}
+		archs = append(archs, *goarch)
+	}
+	if strings.HasSuffix(*output, ".spk") && (len(archs) > 1 || len(dsms) > 1) {
```

Comment:
instead of looking at *output filename, how about requiring that *output be an existing directory if len of either > 1?

Reply:
Done